internal/kldevents/eventstream.go

```diff
@@ -328,10 +328,17 @@ func (a *eventStream) isBlocked() bool {
 // eventPoller checks every few seconds against the ethereum node for any
 // new events on the subscriptions that are registered for this stream
 func (a *eventStream) eventPoller() {
-	defer a.updateWG.Done()
-
 	ctx := kldauth.NewSystemAuthContext()
 
+	defer func() {
+		a.updateWG.Done()
+		// Mark all subscriptions stale, so they will re-start from the checkpoint if/when we re-run the poller
+		subs := a.sm.subscriptionsForStream(a.spec.ID)
+		for _, sub := range subs {
+			sub.markFilterStale(ctx, true)
```

Comment:
There's one thing to be aware of, now that we are doing a few extra things in this deferred function (which potentially involve RPC calls to uninstall filters). Basically what I was thinking is that because deferred functions are stacked, we signal the event poller as done first up, then signal the wait group and then go through marking all the subscriptions stale. It seems like depending on the number of filters that exist in the stream, uninstalling filters could take "some" time. Not sure if it will cause any issues when the stream is restarted soon after (I'm assuming that the filters are re-created).

Reply:
aha, that's a great point. Will refactor

Reply:
I've made the update 👍